### PR TITLE
[23.05]: sifiveu: backport RISC-V target

### PR DESCRIPTION
**Pull request by wigyori**
This PR adds support for the SiFive U-based, Linux-capable RISC-V SoCs.

Supported SoCs:
 - SiFive FU540
 - SiFive FU740

Supported hardware:
 - HiFive Unleashed (FU540)

CPU: SiFive FU540 quad-core RISC-V (U54, RV64IMAFDC or RV64GC)
Memory: 8Gb
Ethernet: 1x 10/100/1000

 - HiFive Unmatched (FU740)

CPU: SiFive FU740 quad-core RISC-V (U74, RV64IMAFDCB or RV64GCB)
Memory: 16Gb
Ethernet: 1x 10/100/1000
USB: 4x USB 3.2
PCIe:
1x PCIe Gen3 x8
1x M.2 key M (PCIe x4)
1x M.2 Key E (PCIe x1 / USB2.0)

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

### CODE REVIEW

**Comment by hauke** [comment]
Please add '(cherry picked from commit ..'  to every commit. git is doing it automatically when you use 'git chery-pick -x '.

File: tools/firmware-utils/patches/001-add-sifiveu-guid-types.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
+diff -ruN firmware-utils-2022-02-28-002cfaf0.old/src/ptgen.c firmware-utils-2022-02-28-002cfaf0/src/ptgen.c
+--- firmware-utils-2022-02-28-002cfaf0.old/src/ptgen.c	2022-04-23 19:02:07.307896842 +0200
++++ firmware-utils-2022-02-28-002cfaf0/src/ptgen.c	2022-04-22 18:48:54.477970950 +0200
```
Comment (hauke) [inline tools/firmware-utils/patches/001-add-sifiveu-guid-types.patch:3]: Please add the code directly to the firmwae-utils project here too:
https://git.openwrt.org/?p=project/firmware-utils.git;a=summary

It is fine if you add it now as a patch, but when someone updates the firmwae-utils  project this should got away.

**Review by hauke — APPROVED** [review]

**Comment by hauke** [comment]
Acked-by: Hauke Mehrtens <hauke@hauke-m.de>

**Review by ynezz — APPROVED** [review]
Nice, thanks!

**Comment by ynezz** [comment]
Merged in 4a281a778999776

**Pull request closed without merging**